test/framework/clusterctl/repository.go

```diff
@@ -159,16 +159,16 @@ func CreateRepository(ctx context.Context, input CreateRepositoryInput) string {
 	return clusterctlConfigFile.Path
 }
 
-// copyAndAmendClusterctlConfigInput is the input for copyAndAmendClusterctlConfig.
-type copyAndAmendClusterctlConfigInput struct {
+// CopyAndAmendClusterctlConfigInput is the input for copyAndAmendClusterctlConfig.
+type CopyAndAmendClusterctlConfigInput struct {
 	ClusterctlConfigPath string
 	OutputPath           string
 	Variables            map[string]string
 }
 
-// copyAndAmendClusterctlConfig copies the clusterctl-config from ClusterctlConfigPath to
+// CopyAndAmendClusterctlConfig copies the clusterctl-config from ClusterctlConfigPath to
 // OutputPath and adds the given Variables.
-func copyAndAmendClusterctlConfig(_ context.Context, input copyAndAmendClusterctlConfigInput) {
+func CopyAndAmendClusterctlConfig(_ context.Context, input CopyAndAmendClusterctlConfigInput) {
```

Comment:
I would suggest to rewrite this func internally to 0 Expects and to return proper errors.

A util func that just fails internally is a pain